client/app/account/billing/common/renew-helper.service.js

```diff
@@ -1,65 +1,74 @@
-angular.module("Billing.services").service("BillingrenewHelper", function ($filter, $translate) {
-    "use strict";
-
-    this.getRenewDateFormated = (service) => {
-        if (!service) {
-            return "";
-        } else if (this.serviceHasAutomaticRenew(service)) {
-            return _.capitalize($filter("date")(service.expiration, "MMMM yyyy"));
+angular
+    .module("Billing.services")
+    .service("billingRenewHelper", class BillingrenewHelper {
+        constructor ($filter, $translate) {
+            this.$filter = $filter;
+            this.$translate = $translate;
         }
 
-        let translationId = "autorenew_service_expiration_date";
-        if (moment().isAfter(moment(service.expiration))) {
-            translationId = "autorenew_service_after_expiration_date";
-        }
-        return $translate.instant(translationId, { t0: $filter("date")(service.expiration, "mediumDate") });
-    };
-
-    this.getRenewLabel = (service) => {
-        if (service === "0") {
-            return $translate.instant("autorenew_service_renew_0");
-        }
+        getRenewDateFormated (service) {
+            if (!service) {
+                return "";
+            } else if (this.serviceHasAutomaticRenew(service)) {
+                return _.capitalize(this.$filter("date")(service.expiration, "MMMM yyyy"));
```

Comment:
What do you think about setting `MMMM yyyy` manually?

Reply:
Alright, alright, I'll refactor this class 😞